internal/ecgit/ecgit.go

```diff
@@ -0,0 +1,306 @@
+// Copyright 2022 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package ecgit
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/plumbing"
+	"github.com/go-git/go-git/v5/plumbing/object"
+	"github.com/go-git/go-git/v5/plumbing/transport/http"
+	"github.com/google/go-github/v45/github"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
+	giturls "github.com/whilp/git-urls"
+	"golang.org/x/oauth2"
+
+	"github.com/hacbs-contract/ec-cli/internal/utils"
+)
+
+var CloneRepoWithAuth = cloneRepoWithAuth
```

Comment:
Why not just name `cloneRepoWithAuth` as `CloneRepoWithAuth`?

Reply:
My thought was that in testing we could set `CloneRepoWithAuth` to some stub or mock function.